src/modules/mavlink/mavlink_main.cpp

```diff
@@ -603,7 +607,7 @@ Mavlink::mavlink_open_uart(int baud, const char *uart_name, bool force_flow_cont
 				return -1;
 			}
 
-			px4_usleep(100000);
+			px4_usleep(_uart_fd == ENOTCONN ? 1000000 : 100000); // ENOTCONN means that the USB device is not yet connected.
```

Comment:
I don't think this is correct because `open` is supposed to return -1 on error and set `errno`. Therefore, you are probably always ending up in the else case.

Reply:
It is not correct.

@mcsauder  this is the code snippet I gave you:

```
          int errcode = errno;
          /* ENOTCONN means that the USB device is not yet connected */
              usleep(errcode == ENOTCONN ? 1000 000 :  100000 );
```

See https://github.com/PX4/NuttX-apps/blob/7db2a352fb19438bb966e419c61e55092056d385/examples/usbserial/usbserial_main.c#L266-L287 for a complete example

Reply:
Thanks @davids5 , my mistake, I mistakenly thought you were giving me a generic example to follow.  Fixed and pushed up.